internal/handlers/handler_verify.go

```diff
@@ -171,7 +171,10 @@ func verifySessionCookie(ctx *middlewares.AutheliaCtx, targetURL *url.URL, userS
 				return "", "", nil, nil, authentication.NotAuthenticated, fmt.Errorf("unable to destroy user session after long inactivity: %s", err)
 			}
 
-			return userSession.Username, userSession.DisplayName, userSession.Groups, userSession.Emails, authentication.NotAuthenticated, fmt.Errorf("User %s has been inactive for too long", userSession.Username)
+			ctx.Logger.Warnf("User %s has been inactive for too long", userSession.Username)
+
+			// Fixes #3587.
+			return userSession.Username, userSession.DisplayName, userSession.Groups, userSession.Emails, authentication.NotAuthenticated, nil
```

Comment:
```suggestion
			return "", "", nil, nil, authentication.NotAuthenticated, nil
```

Sorry I was wrong about this. instead change: https://github.com/authelia/authelia/blob/2781a199408fc127088f860ad368bccd7627982c/internal/handlers/handler_verify.go#L266-L269

So that it  doesn't ignore empty usernames (username argument can be removed as part of this change).